### ISSUE TITLE
falsely reports links to custom pages as invalid

### ISSUE THREAD

**Issue by lschierer** [HiDeoo/starlight-links-validator#39]
### Describe the bug

if I create custom pages inside ```projectroot/src/pages``` and then link to those pages from within a .md or .mdx file in a content collection, the link validator will declare that link an invalid link, despite the fact that it works from astro dev and when published to a dev stage. 

### To Reproduce

1. go to https://stackblitz.com/edit/withastro-starlight-vayr1z?file=src%2Fpages%2Fsomething%2Ftestpage.astro
2. click testpage on the initial splash page. it is under the Next steps heading.
3. See that link works.
4. control c the astro dev session
5. ```npm run build```
6. see that the link validator failed on that link

### Expected behavior

it is a valid link, it should have passed the link validator test. 

### How often does this bug happen?

Every time

### System Info

- OSX 14.3.1
- Chrome
- pnpm 9.0.6
- astro 4.7.0
- starlight 0.21.5
- starlight-links-validator 0.8.0

**Comment by HiDeoo** [comment]
Thanks for your report :raised_hands:

I'm a bit torn on this one. The main issue with custom pages is that I cannot easily and reliably gather as much informations as for Starlight pages, e.g. headings, i18n, fallbacks, etc.

If the validation cannot be performed as reliably as for Starlight pages, my first thought is that maybe they should not be validated at all and expect users to [`exclude`](https://starlight-links-validator.vercel.app/configuration/#exclude) them from the validation process. :thinking:

I still need to think about it a bit more but I'm open to suggestions and feedback on this topic.

**Comment by lschierer** [comment]
For my use case, it is sufficient that the link is _valid_, ie not a 404.   I understand that many people have far more complex needs.  I'm also fairly straight forward in that I'm not using any i18n.   So the extra checks on headings, fallbacks and such are unnecessary for me, but I understand their value on sites with wider audiences.  

If you can tell that it matches a route that Astro knows about, then rather than making me exclude each page, can I have an option to exclude all custom pages? 
If you cannot tell when it matches a route or not, then the exclude option may be best, but a documentation update is certainly a must to explain that custom pages will result in validation failures. 

Thanks!

**Comment by HiDeoo** [comment]
Yeah, I still want to play a bit with the code and make sure of everything but I think I may not be able to reliably tell wether a page would be a custom page or let's say a fallback page for example generated by Starlight.

If I discover I can, I'll explore the idea. If not, I think updating the docs like you mentioned will be the best way to go at least for now.

**Comment by HiDeoo** [comment]
I played a bit with the idea but ended updating, like you suggested, the [`exclude`](https://starlight-links-validator.vercel.app/configuration/#exclude)  documentation as I did not find any reliable way for this specific case at the moment.

I made sure to link to the Starlight [custom pages](https://starlight.astro.build/guides/pages/#custom-pages) documentation and that searching for "page" or "custom page" would lead to the proper configuration option.

I'm thinking this may be a topic re-opened in the future, maybe with some changes to Astro’s route manifest for example or related changes in this area so we can maybe revisit this in the future and provide a better solution.

Thanks again for your feedback.
